### PR TITLE
1) some bugs in JSP's are changed

### PR DESCRIPTION
**Pull request by vakhonin**
2) adding prolongation (need add method in subscription service
for right working)

### CODE REVIEW

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 1]
```diff
@@ -136,4 +141,29 @@ public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1")
         return "redirect:" + USER;
     }
 
-}
+
+
+
+    @PostMapping(value = "/prolongate")
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: To PathUtils

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 2]
```diff
@@ -136,4 +141,29 @@ public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1")
...
+    public @ResponseBody String prolongateSubscription(@RequestParam int duration) {
+
+
+//        Subscription subscription = subscriptionService.getSubscriptionById(id);
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: What is the comments ?

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 3]
```diff
@@ -136,4 +141,29 @@ public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1")
...
+        // need to get id of subscription from hidden input
+
+
+
+        // work with 1 Igor subscription
+
+        Subscription subscription = subscriptionService.getSubscriptionsByAccountId(2, 0, 10).get(0);
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: What do this numbers mean ?

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 4]
```diff
@@ -136,4 +141,29 @@ public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1")
...
+        subscriptionService.prolongSubscription(subscription, duration);
+        subscription = subscriptionService.getSubscriptionsByAccountId(2, 0, 10).get(0);
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: What do these numbers mean ?

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 5]
```diff
@@ -136,4 +141,29 @@ public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1")
...
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: String to constant

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 6]
```diff
@@ -136,4 +141,29 @@ public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1")
...
+
+        String result = subscription.getEndDate().format(formatter).toString();
+
+        return result;
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: Inline

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 7]
```diff
@@ -136,4 +141,29 @@ public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1")
...
+    }
+
+
+}
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: why so much empty lines between code blocks?
do not store comments in git, comments mean that you failed to code something in readable/maintainable manner and left a comments, remember that there is always a refactor stage, where you do refactor a working code into readable/maintainable

File: src/main/webapp/WEB-INF/jsp/user-page.jsp [hunk 8]
```diff
@@ -48,17 +51,63 @@
...
+                }}).done(function (newDate) {
+                    // jQuery('.ajax').parent().parent().find('.new_date').text(newDate);
+                    jQuery(btn).parent().parent().find('.new_date').text(newDate);
+                }
+            ).fail(function (e) {
+                console.log(e);
+            });
+        }
+    });
+</script>
```
Comment (Jebsuz) [inline src/main/webapp/WEB-INF/jsp/user-page.jsp]: extract this code to a file and import in here via `<script type="text/javascript" src=file.js />`
HTML page rules applies to JSP as well, do not store an inline javascript on page, store it in separate file with meaningful name and import it on pages, thus making the code available for re-use and easy maintain

**Pull request merged**